build-scripts/VsDevCmd.cmake

```diff
@@ -0,0 +1,101 @@
+#[=======================================================================[
+
+VsDevCmd.cmake
+--------------
+
+Run VsDevCmd.bat and extracts environment variables it changes.
+Inject them into the global scope and _MSVC_DEVENV for
+CMakeUserPreset.json at a later step.
+
+#]=======================================================================]
+
+if("$ENV{VSCMD_VER}" STREQUAL "")
+    # This cmake process is running under VsDevCmd.bat or VS IDE GUI
+    # Avoid to run VsDevCmd.bat twice
+
+    if ("$ENV{DevEnvDir}" STREQUAL "")
+        # Use Community Edition when not specified
+        set(ENV{DevEnvDir} "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/")
```

Comment:
Maybe use `vswhere`? https://github.com/microsoft/vswhere

Reply:
Done in fdb7e95